File: tests/fixtures/codegen/inputs/pointers-expected/typescript/runtime/queries/items/ReturningMany.ts
import type { ReturningManyParams, ReturningManyRow } from "../../../types/queries/items/ReturningMany.js";
export type { ReturningManyParams, ReturningManyRow } from "../../../types/queries/items/ReturningMany.js";
import * as pgsid from "./pgsid/queryable.js";
export { QueryValidationError } from "./pgsid/queryable.js";
import * as _jsonSchemas from "../../jsonschemas/index.js";
export const returningManySql = "UPDATE items SET payload = $1 RETURNING payload;";
export function isReturningManyPayload(value: unknown): value is ReturningManyRow["payload"] {
    return _jsonSchemas.isEvent(value);
}
export function validateReturningManyPayload(value: unknown): {
    valid: boolean;
    issues: {
        path: readonly (string | number)[];
        keyword: string;
        expected: unknown;
        received: string;
        message: string;
    }[];
} {
    return _jsonSchemas.validateEvent(value);
}
export function isReturningManyParamsPayloadPublicItemsPayload(value: unknown): value is ReturningManyParams["payload"] {
    return _jsonSchemas.isEvent(value);
}
export function validateReturningManyParamsPayloadPublicItemsPayload(value: unknown): {
    valid: boolean;
    issues: {
        path: readonly (string | number)[];
        keyword: string;
        expected: unknown;
        received: string;
        message: string;
    }[];
} {
    return _jsonSchemas.validateEvent(value);
}
export async function returningMany(db: pgsid.Queryable, params: ReturningManyParams): Promise<ReturningManyRow[]> {
    const jsonInput1 = JSON.stringify(params["payload"]);
    if (jsonInput1 === undefined)
        throw new pgsid.QueryValidationError("ReturningMany", "payload", [{ path: [], keyword: "json", expected: "JSON value", received: "undefined", message: "Value has no JSON representation" }]);
    const jsonInput1Value = jsonInput1 === null ? null : JSON.parse(jsonInput1);
    if (jsonInput1 !== null) {
        if (!isReturningManyParamsPayloadPublicItemsPayload(jsonInput1Value))
            throw new pgsid.QueryValidationError("ReturningMany", "payload", validateReturningManyParamsPayloadPublicItemsPayload(jsonInput1Value).issues);
    }
    const result = await db.query(returningManySql, [jsonInput1]);
    return result.rows.map(row => {
        if (!isReturningManyPayload(row["payload"]))
            throw new pgsid.QueryValidationError("ReturningMany", "payload", validateReturningManyPayload(row["payload"]).issues);
        return row as ReturningManyRow;
    });
}

File: tests/fixtures/codegen/inputs/structs-expected/queries/items/returningmany.go
package items

import (
	"context"
	jsonschemas "example.com/pgsid-validation/generated/jsonschemas"
	pgsidpgx "example.com/pgsid-validation/generated/queries/pgsid/pgx"
)

const ReturningManySQL = "UPDATE items SET payload = $1 RETURNING payload;"

type ReturningManyParams struct {
	Payload jsonschemas.Event `db:"payload"`
}
type ReturningManyRow struct {
	Payload jsonschemas.Event `db:"payload"`
}

func (q *Queries) ReturningMany(ctx context.Context, params ReturningManyParams) ([]ReturningManyRow, error) {
	jsonInput1, err := pgsidpgx.ValidateJSONInput(params.Payload, "pgsid:///jsonschemas/Event.json#", "ReturningMany", "payload", false)
	if err != nil {
		return nil, err
	}
	rows, err := q.db.Query(ctx, ReturningManySQL, jsonInput1)
	if err != nil {
		return nil, err
	}
	defer rows.Close()
	items := make([]ReturningManyRow, 0)
	for rows.Next() {
		var row ReturningManyRow
		if err := rows.Scan(pgsidpgx.ScanTargets(pgsidpgx.ValidatedJSON(&row.Payload, "pgsid:///jsonschemas/Event.json#", "ReturningMany", "payload", false))); err != nil {
			return nil, err
		}
		items = append(items, row)
	}
	if err := rows.Err(); err != nil {
		return nil, err
	}
	return items, nil
}
